_inc/lib/class.media-summary.php

```diff
@@ -63,11 +63,34 @@ static function get( $post_id, $blog_id = 0, $args = array() ) {
 				switch ( $type ) {
 					case 'wpvideo':
 						if ( 0 == $return['count']['video'] ) {
-							$return['type'] = 'video';
-							$return['video'] = esc_url_raw( 'http://s0.videopress.com/player.swf?guid=' . $extract['shortcode']['wpvideo']['id'][0] . '&isDynamicSeeking=true' );
-							$return['image'] = self::get_video_poster( 'videopress', $extract['shortcode']['wpvideo']['id'][0] );
-							$return['secure']['video'] = preg_replace( '@http://[^\.]+.videopress.com/@', 'https://v0.wordpress.com/', $return['video'] );
-							$return['secure']['image'] = str_replace( 'http://videos.videopress.com', 'https://videos.files.wordpress.com', $return['image'] );
+							$guid = $extract['shortcode']['wpvideo']['id'][0];
+							$video_info = videopress_get_video_details( $guid );
```

Comment:
I'm concerned about the implications of calling to this on every page load. We should definitely at least do some sort of caching routine. Though, long term, it as we start to merge more and more VideoPress things into the local library, we may find that we're able to do this call locally and rely on Jetpack sync in order to keep that data up to date.